tests/unit/deferred.py

```diff
@@ -0,0 +1,201 @@
+        }
+
+        id = db.Column(db.Integer,
+                       db.ForeignKey('switch.id'),
+                       primary_key=True)
+        hostname = db.Column(db.String, nullable=False)
+        username = db.Column(db.String, nullable=False)
+        password = db.Column(db.String, nullable=False)
+        last_count = None
+
+        @staticmethod
+        def validate(kwargs):
+            pass
+
+        def session(self):
+            return self
+
+        def disconnect(self):
+            pass
+
+        def modify_port(self, port, channel, network_id):
+
+            # setup a different connection to database so that this method does
+            # not see uncommited changes by `apply_networking`
+
+            local_app = Flask(__name__.split('.')[0])
+            uri = config.cfg.get('database', 'uri')
+            local_app.config.update(SQLALCHEMY_TRACK_MODIFICATIONS=False)
+            local_app.config.update(SQLALCHEMY_DATABASE_URI=uri)
+            local_db = SQLAlchemy(local_app)
+
+            current_count = local_db.session \
+                .query(model.NetworkingAction).count()
+
+            local_db.session.commit()
+
+            if self.last_count is None:
+                self.last_count = current_count
+            else:
+                assert current_count == self.last_count - 1, \
+                  "network daemon did not commit previous change!"
+                self.last_count = current_count
+
+    DeferredTestSwitch_.__name__ = 'DeferredTestSwitch'
+    DeferredTestSwitch = DeferredTestSwitch_
+
+
+@pytest.fixture()
+def switch(_deferred_test_switch_class):
+    return DeferredTestSwitch(
+        label='switch',
+        hostname='http://example.com',
+        username='admin',
+        password='admin',
+    ).session()
+
+
+@pytest.fixture()
+def nic1():
+    from hil.ext.obm.mock import MockObm
+    return model.Nic(
+        model.Node(
+            label='node-99',
+            obm=MockObm(
+                type="http://schema.massopencloud.org/haas/v0/obm/mock",
+                host="ipmihost",
+                user="root",
+                password="tapeworm")),
+        'ipmi',
+        '00:11:22:33:44:55')
+
+
+@pytest.fixture()
+def nic2():
+    from hil.ext.obm.mock import MockObm
+    return model.Nic(
+        model.Node(
+            label='node-98',
+            obm=MockObm(
+                type="http://schema.massopencloud.org/haas/v0/obm/mock",
+                host="ipmihost",
+                user="root",
+                password="tapeworm")),
+        'ipmi',
+        '00:11:22:33:44:55')
+
+
+@pytest.fixture()
+def network():
+    project = model.Project('anvil-nextgen')
+    return model.Network(project, [project], True, '102', 'hammernet')
+
+pytestmark = pytest.mark.usefixtures('configure',
+                                     'fail_on_log_warnings')
+
+
+def test_apply_networking(switch, nic1, nic2, network, fresh_database):
```

Comment:
Would it be possible to add another test that ensures that if action2 fails, action1 successfully was committed?